src/ObjectStateMutations.ts

```diff
@@ -191,7 +191,7 @@ export function commitServerChanges(
       typeof val === 'object' &&
       !Array.isArray(val) &&
       Object.keys(val).length > 0 &&
-      Object.keys(val).some(k => !isNaN(parseInt(k))) &&
+      Object.keys(val).every(k => k === String(Number(k)) && Number.isInteger(Number(k))) &&
```

Comment:
Add another check for dot notation arrays.

A dot notation on array looks like `items.0.count` just check for a dot and an index.

The result from the server would return something like

{ items: { '0': { count: 20 } } }

Reply:
@mtrezza Can you make dot notation on arrays experimental?

`CoreManager.set('ENABLE_DOT_NOTATION_ARRAY', true)`